projects/app-lob/src/app/grid-finjs/main.component.ts

```diff
@@ -86,7 +86,14 @@ export class FinJSDemoComponent implements OnDestroy {
     public setChartData(args: Stock[]): void {
         this.chartData = [];
         args.forEach(row => {
-            this.chartData.push(row);
+            this.finGrid.grid.data.filter(data => {

```

Comment:
There is a forEach loop. and inside it there is filter, which again loops through all data records. Not a viable solution. This increases complexity and iterations a lot. Must be changed.